site/en/configure/build-performance-breakdown.md

```diff
@@ -0,0 +1,231 @@
+     before further drilling down with
+     [`--skyframe_state`](/docs/reference/command-line-reference#flag--skyframe_state).
+
+4. `ActionSummary.actions_executed`: the number of actions executed, a
+  regression directly represents more work in executing these actions.
+   - The [BEP](/docs/remote/bep) writes out the action statistics
+     `ActionData` that shows the most executed action types. By default, it
+     collects the top 20 action types, but you can pass in the
+     [`--experimental_record_metrics_for_all_mnemonics`](/docs/reference/command-line-reference#flag--experimental_record_metrics_for_all_mnemonics)
+     to collect this data for all action types that were executed.
+   - This should help you to figure out what kind of actions were executed
+     (additionally).
+
+5. `BuildGraphSummary.outputArtifactCount`: the number of artifacts created by
+  the executed actions.
+   - If the number of actions executed did not increase, then it is likely that
+     a rule implementation was changed.
+
+
+These metrics are all affected by the state of the local cache, hence you will
+want to ensure that the builds you extract these metrics from are
+**clean builds**.
+
+We have noted that a regression in any of these metrics can be accompanied by
+regressions in wall time, cpu time and memory usage.
+
+### Usage of local resources
+
+Bazel consumes a variety of resources on your local machine (both for analyzing
+the build graph and driving the execution, and for running local actions), this
+can affect the performance / availability of your machine in performing the
+build, and also other tasks.
+
+#### Time spent
+
+Perhaps the metrics most susceptible to noise (and can vary greatly from build
+to build) is time; in particular - wall time, cpu time and system time. You can
+use [bazel-bench](https://github.com/bazelbuild/bazel-bench) to get
+a benchmark for these metrics, and with a sufficient number of `--runs`, you can
+increase the statistical significance of your measurement.
+
+- **Wall time** is the real world time elapsed.
+   - If _only_ wall time regresses, we suggest collecting a
+     [JSON trace profile](/docs/configure/json-trace-profile) and looking for
+     differences. Otherwise, it would likely be more efficient to investigate
+     other regressed metrics as they could have affected the wall time.
+
+- **CPU time** is the time spent by the CPU executing user code.
+   - If the CPU time regresses across two project commits, we suggest collecting
+     a Starlark CPU profile. You should probably also use `--nobuild` to
+     restrict the build to the analysis phase since that is where most of the
+     CPU heavy work is done.
+
+- System time is the time spent by the CPU in the kernel.
+   - If system time regresses, it is mostly correlated with I/O when Bazel reads
+     files from your file system.
+
+#### System-wide load profiling
+
+Using the `--experimental_collect_load_average_in_profiler` flag, the
```

Comment:
I could not find an entry for this flag on https://bazel.build/reference/command-line-reference - is this a hidden flag, or a new flag in 6.0?

Reply:
https://github.com/bazelbuild/bazel/blob/release-6.0.0/src/main/java/com/google/devtools/build/lib/runtime/CommonCommandOptions.java#L306-L312

Reply:
Thanks. Linking to the code seems too prone to change, unless we include the commit id. Should we simply not linkify this flag? Are there plans to "unhide" the flag in the future?

Reply:
I think most experimental flags are hidden by default (though I don't like that practice). I think linking to the code is _fine_ as long as it's a permalink pointing to a tag (so now what I linked, which is a branch).

@larsrc-google Thoughts on making this flag publicly documented so we can link to the docs?

Reply:
> I think most experimental flags are hidden by default (though I don't like that practice). I think linking to the code is _fine_ as long as it's a permalink pointing to a tag (so now what I linked, which is a branch).

Done.

Reply:
Ok, I adjusted the links (which are currently still broken, but will 6.0.0 will hopefully land soon).